src/index.ts

```diff
@@ -49,6 +49,7 @@ export async function run() {
   info(`Event name: ${context.eventName}`);
   switch (context.eventName) {
     case 'pull_request':
+    case 'pull_request_target':
```

Comment:
This event is run in the context of the PR's base commit instead of its head commit. That means it won't have the changes in the PR. When the Action tries to determine the diff in the PR, there will be no diff. 

https://stackoverflow.com/questions/74957218/what-is-the-difference-between-pull-request-and-pull-request-target-event-in-git

Can you tell me more about your need of running this Action with this event?

Reply:
Sure! The `pull_request` event lacks access to secrets when a pull request is made from a forked repository. That makes the GitHub token inaccessible so this action does not work in that case. This is a pretty big problem for open source projects where the majority of PRs come from forks. The solution I found was to change the workflow to target the `pull_request_target` event and manually specify the PR sha as the ref in the checkout step.

Reply:
That makes sense.